### DISCUSSION TITLE
[Feature Request] diesel_cli(print-schema): Allow derives for custom (SQL) types to be configurable

### DISCUSSION THREAD

**Discussion by SanchithHegde** [diesel-rs/diesel#3306] [Ideas]
Since `diesel_cli` 2.0 auto-generates the definitions for custom SQL types ([this snippet](https://github.com/diesel-rs/diesel/blob/1327fba7b2d0419b631e4ff24e4ca430d62fc2fe/examples/postgres/custom_types/src/schema.rs#L3-L7)) when `print-schema` (and hence `migration run`) command is run, it removes a significant amount of our code that had to be manually written by hand and maintained for every enum that we use in our database (Postgres). However, the type definitions generated lack some derives that are required for our code to compile (`Debug` and `QueryId` in our case).

Could we please have the list of traits/implementations derived, be configurable via a field in the `diesel.toml` file?

1. I'm thinking we can have a field called `sql_type_derives` (I'd appreciate better name suggestions), that appends the list of derives to what exists currently (`SqlType`). 
2. Maybe even derive `Debug` by default?

Open to others' opinions and solutions for this problem. I can open a PR if necessary.

---

I've gone through the code, this specific line would have to be updated:
https://github.com/diesel-rs/diesel/blob/1327fba7b2d0419b631e4ff24e4ca430d62fc2fe/diesel_cli/src/print_schema.rs#L278

**Comment by weiznich** [comment]
See https://github.com/diesel-rs/diesel/pull/3297#issuecomment-1232809465
As mentioned there: PR's welcome

**Comment by SanchithHegde** [comment]
Addressed in PR #3308.
